src/main/java/com/google/firebase/FirebaseOptions.java

```diff
@@ -71,6 +76,26 @@ FirebaseCredential getCredential() {
     return databaseAuthVariableOverride;
   }
 
+  /**
+   * Returns the HttpTransport used to call remote HTTP endpoints.
+   *
+   * @return A Google API client HttpTransport instance.
+   */
+  @NonNull
+  public HttpTransport getHttpTransport() {
+    return httpTransport;
+  }
+
+  /**
+   * Returns the JsonFactory used to parse JSON when calling remote HTTP endpoints.
+   *
+   * @return A Google API client JsonFactory instance.
+   */
+  @NonNull
+  public JsonFactory getJsonFactory() {
+    return jsonFactory;
+  }
+
   @Override
   public boolean equals(Object obj) {
     if (!(obj instanceof FirebaseOptions)) {
```

Comment:
It looks like these were implemented to support some persistence requirements, which are not really used in the SDK anymore. In the future I can see us adding even more options to this class, and having to implement all these methods each time we make a change is a bit of a problem. WDYT? Should we just get rid of the implementations?

`FirebaseApp` also implements these methods, but at least it performs all comparisons based on the name of the app.